File: src/main/java/com/xuxiaocheng/WListClient/Server/FailureReason.java
package com.xuxiaocheng.WListClient.Server;

import org.jetbrains.annotations.NotNull;

public enum FailureReason {
    InvalidFilename,
    DuplicatePolicyError,
    ExceedMaxSize,
    NoSuchFile,
    Others,
    ;

    public static @NotNull String handleFailureReason(final @NotNull FailureReason reason) {
        return switch (reason) {
            case InvalidFilename -> "Invalid filename.";
            case DuplicatePolicyError -> "ERROR by duplicate policy.";
            case ExceedMaxSize -> "Exceed max size per file.";
            case NoSuchFile -> "No such file.";
            case Others -> "Failure, unknown reason.";
        };
    }
}

File: src/handlers/failure_reason.rs
use std::fmt::{Display, Formatter};

pub enum FailureReason {
    InvalidFilename,
    DuplicatePolicyError,
    ExceedMaxSize,
    NoSuchFile,
    Others,
}

impl Display for FailureReason {
    fn fmt(&self, f: &mut Formatter<'_>) -> std::fmt::Result {
        write!(f, "{}", String::from(self))
    }
}

impl From<&FailureReason> for String {
    fn from(value: &FailureReason) -> Self {
        String::from(match value {
            FailureReason::InvalidFilename => "Invalid filename.",
            FailureReason::DuplicatePolicyError => "ERROR by duplicate policy.",
            FailureReason::ExceedMaxSize => "Exceed max size per file.",
            FailureReason::NoSuchFile => "No such file.",
            FailureReason::Others => "Failure, unknown reason.",
        })
    }
}
